server.py

```diff
@@ -15,33 +17,65 @@ def setup():
     )
     sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     sock.bind(addr)
-    sock.listen(1)
+    sock.listen(2)
     return sock
 
 
 def response_ok():
     """Return Header and Body information for Response 200."""
-    response_headers = 'HTTP/1.1 200 OK\nContent-Type: text/html'
-    response_body = '<html><body><h1></h1>'\
-        '<p>All good here, captain.</p></body></html>'
+    response_headers = (
+        b'HTTP/1.1 200 OK\r\n'
+        + date +
+        b'Content-Type: text/html\r\n'
+        b'Content-Length:\r\n')
+
+    response_body = (
+        b'<html><body>'
+        b'<p>All good here, captain.</p>'
+        b'</body></html>')
 
     return response_headers + response_body
 
 
-def response_error():
-    """Return Header and Body information for Response 500."""
-    response_headers = 'HTTP/1.1 500 Internal Server Error\n' +\
-        'Content-Type: text/html'
-    response_body = '<html><body><h1></h1>'\
-        '<p>Your server no bueno.</p></body></html>'
+def response_error(header, text):
+    """Return Header and Body information for three types of errors"""
+
+    response_headers = (
+        header +
+        date +
```

Comment:
since date is a module-level constant, there will only ever be one date for all the responses this server sends, until it is stopped and restarted.  Then a new date will happen.